storage/remote/queue_manager.go

```diff
@@ -193,9 +193,11 @@ type QueueManager struct {
 	client         StorageClient
 	watcher        *WALWatcher
 
-	seriesLabels         map[uint64]labels.Labels
-	seriesSegmentIndexes map[uint64]int
-	droppedSeries        map[uint64]struct{}
+	seriesLabels            map[uint64]labels.Labels
+	seriesLabelsMtx         sync.Mutex
```

Comment:
`sync.RWMutex` would be better?

Reply:
I do not believe it is possible for multiple readers to be accessing any of these maps concurrently (there is only ever one `Append` call happening at a time), so a `sync.RWMutex` is unneeded.